e2e/ceph_user.go

```diff
@@ -0,0 +1,95 @@
+package e2e
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+// #nosec because of the word `Secret`
+const (
+	// ceph user names
+	keyringRBDProvisionerUsername          = "cephcsi-rbd-provisioner"
+	keyringRBDNodePluginUsername           = "cephcsi-rbd-node"
+	keyringRBDNamespaceProvisionerUsername = "cephcsi-rbd-ns-provisioner"
+	keyringRBDNamespaceNodePluginUsername  = "cephcsi-rbd-ns-node"
+	keyringCephFSProvisionerUsername       = "cephcsi-cephfs-provisioner"
+	keyringCephFSNodePluginUsername        = "cephcsi-cephfs-node"
+	// secret names
+	rbdNodePluginSecretName           = "cephcsi-rbd-node"
+	rbdProvisionerSecretName          = "cephcsi-rbd-provisioner"
+	rbdNamespaceNodePluginSecretName  = "cephcsi-rbd-ns-node"
+	rbdNamespaceProvisionerSecretName = "cephcsi-rbd-ns-provisioner"
+	cephFSNodePluginSecretName        = "cephcsi-cephfs-node"
+	cephFSProvisionerSecretName       = "cephcsi-cephfs-provisioner"
+)
+
+func rbdNodePluginCaps(pool, rbdNamespace string) []string {
```

Comment:
ISeparate it out as we can have different caps for provisioner or nodeplugin

Reply:
@Madhu-1 ok. fine . We can have 2 functions with same body. However the caps can be defined with `osd profile rbd ` in it , so that the return  for first namespace check dont need any slice operation in it.

Reply:
splice operation is useful when we add more caps to the users in the future. if you don't want it I can return it from there but still, I need to append and return it. I might be missing something can you please help me with some pseudo code?

Reply:
```rbdNodePluginCapsSlice := []string{ "mon", "'profile rbd'", "mgr", "'allow rw'", "osd", "'profile rbd'", }
rbdProvisionerCapsSlice := []string{ "mon", "'profile rbd'", "mgr", "'allow rw'", "osd", "'profile rbd'", }
rbdNodePluginCaps(pool,rbdnamespace) {
 if rbdnamespace {
  return append(rbdNodePluginCapsSlice , fmt.Sprintf("osd 'profile rbd pool=%s namespace=%s'", pool, rbdNamespace))
}
return rbdNodePluginCapsSlice 
}
same way rbdProvisionerPluginCaps(pool, rbdnamespace){
....
}```

Considering the rbdNodePluginCapsSlice and rbdProvisionerPluginCapsSlice have pretty much same common base, we can avoid 2 slices definition in the top too if needed.

Reply:
I see the only difference I see is declaring a global slice. do we need to do it?
and also I would like to avoid using a single slice even if the caps looks similar for now as we test more restricted users/caps having separate slices makes ease of code maintenance

Reply:
not only packing to a slice global, it also avoid one append operation in both functions.

Reply:
but if the rbdnamespace is passed it will fail because of duplication osd profile as we are appending for an already existing osd in the CapsSlice

Reply:
> but if the rbdnamespace is passed it will fail because of duplication osd profile as we are appending for an already existing osd in the CapsSlice

we dont need that append if namespace is non nill, we can just append `pool` and `namespace`.

Reply:
@humblec do you want to change like below, if yes even need to append twice because `'` at the end is important

```go
rbdNodePluginCapsSlice := []string{ "mon", "'profile rbd'", "mgr", "'allow rw'", "osd", "'profile rbd", }
rbdProvisionerCapsSlice := []string{ "mon", "'profile rbd'", "mgr", "'allow rw'", "osd", "'profile rbd", }

rbdNodePluginCaps(pool,rbdnamespace) {
 if rbdnamespace {
  return append(rbdNodePluginCapsSlice , fmt.Sprintf("pool=%s namespace=%s'", pool, rbdNamespace))
}
// add `'` to the end of profile rbd because "'" is important or else the command will fail.
// removing the last object and re-adding `'` and again appending it because complicated
}
}```

Reply:
@Madhu-1 as mentioned earlier I am fine with either way ( current version or some more optimized as above ), please let me know the choice, so that we can move ahead with that.

Reply:
I would like to go with current approach and change it later if required